src/extractor/extractOpacities.ts

```diff
@@ -0,0 +1,33 @@
+import extractorInterface from '@typings/extractorInterface'
+import { opacityPropertyInterface } from '@typings/propertyObject'
+import { customTokenNode } from '@typings/tokenNodeTypes'
+import { PropertyType } from '@typings/valueTypes'
+import { tokenTypes } from '@config/tokenTypes'
+import roundWithDecimals from '../utilities/roundWithDecimals'
+import { filterByPrefix } from './extractUtilities'
+import { tokenExportKeyType } from '@typings/tokenExportKey'
+import { tokenCategoryType } from '@typings/tokenCategory'
+import config from '@config/config'
+
+const extractOpacities: extractorInterface = (tokenNodes: customTokenNode[], prefixArray: string[]): opacityPropertyInterface[] => {
+  // return as object
+  return tokenNodes.filter(filterByPrefix(prefixArray)).map(node => ({
+    name: node.name,
+    category: 'opacity' as tokenCategoryType,
+    exportKey: tokenTypes.opacity.key as tokenExportKeyType,
+    description: node.description || null,
+    values: {
+      opacity: {
+        value: roundWithDecimals(node.opacity, 2),
```

Comment:
Is node.opacity always between 0 and 1? Otherwise we would need a safeguard to set it to 1 if null or undefined.

Seems like it should be true: https://www.figma.com/plugin-docs/api/Paint/#opacity

Reply:
This one was left unchanged. As you pointed out, Figma nodes won't have values outside of the 0..1 interval. If you really want to be defensive in our coding right here, i'm more than happy to add that, but I personally think it's not adding any value.